dlib/cmake_utils/find_blas.cmake

```diff
@@ -66,16 +66,29 @@ if (UNIX OR MINGW)
    find_package(PkgConfig)
    pkg_check_modules(BLAS_REFERENCE cblas)
    pkg_check_modules(LAPACK_REFERENCE lapack)
-   if (BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND)
+
+   SET(CMAKE_REQUIRED_LIBRARIES "cblas")
+   INCLUDE(CheckFunctionExists)
+   CHECK_FUNCTION_EXISTS(cblas_ddot HAVE_CBLAS)
+   SET(CMAKE_REQUIRED_LIBRARIES "${BLAS_REFERENCE_LDFLAGS}")
```

Comment:
You have to set `CMAKE_REQUIRED_LIBRARIES` **before** you call `CHECK_FUNCTION_EXISTS`.  It's an input to `CHECK_FUNCTION_EXISTS`. Also, per the docs (https://cmake.org/cmake/help/v3.14/module/CheckFunctionExists.html) what you need to do is likely this: 
```
SET(CMAKE_REQUIRED_LINK_OPTIONS "${BLAS_REFERENCE_LDFLAGS}")
SET(CMAKE_REQUIRED_LIBRARIES "cblas")
CHECK_FUNCTION_EXISTS(cblas_ddot HAVE_CBLAS)
```